src/lib/philips.ts

```diff
@@ -417,17 +417,20 @@ export const philipsFz = {
         convert: (model, msg, publish, options, meta) => {
             const buttonLookup: KeyValue = {1: 'button_1', 2: 'button_2', 3: 'button_3', 4: 'button_4', 20: 'dial'};
             const button = buttonLookup[msg.data['button']];
-            const typeLookup: KeyValue = {0: 'press', 1: 'hold', 2: 'press_release', 3: 'hold_release'};
-            const type = typeLookup[msg.data['type']];
             const direction = msg.data['unknown2'] <127 ? 'right' : 'left';
             const time = msg.data['time'];
             const payload: KeyValue = {};
 
             if (button === 'dial') {
+                const typeLookup: KeyValue = {1: 'step', 2: 'rotate'};
+                const type = typeLookup[msg.data['type']];
                 const adjustedTime = direction === 'right' ? time : 256 - time;
                 const dialType = 'rotate';
                 const speed = adjustedTime <= 25 ? 'step' : adjustedTime <= 75 ? 'slow' : 'fast';
                 payload.action = `${button}_${dialType}_${direction}_${speed}`;
```

Comment:
Instead of adding `type`, isn't it more logical to add it here instead of `dialType`?

Reply:
I would say no. `dialType` is hardcoded to `rotate`, while the `type` that button exposes is an info whether is a first message in the action of turning the dial or consequent. Since `action` already interprets `time` value to a enum `dialType` provides only a nicely readable string to the end user, no extra information.

I want to achieve with this PR 2 things:
* Expose the raw data that the dial provides so I can use them in custom functions in node red
* Keep the backwards compatibility so I don't break existing implementations of other people (that means not changing `action`)

The button exposes those 3 attributes and I'm just mapping them to same name. Please let me know if you want further changes on the naming.